app/src/main/java/org/wikipedia/watchlist/WatchlistViewModel.kt

```diff
@@ -170,4 +179,63 @@ class WatchlistViewModel : ViewModel() {
         val types = WatchlistFilterTypes.TYPE_OF_CHANGES_GROUP.filter { includedTypesCodes.contains(it.id) }.map { it.value }
         return types.joinToString(separator = "|")
     }
+
+    companion object {
+        suspend fun watchPageTitle(scope: CoroutineScope, pageTitle: PageTitle, unwatch: Boolean, expiry: WatchlistExpiry = WatchlistExpiry.NEVER,
+                                   isCurrentlyWatched: Boolean? = null, isTalkPage: Boolean? = null): Pair<Boolean, String> {
+            isCurrentlyWatched?.let {
+                if (it) {
+                    WatchlistAnalyticsHelper.logRemovedFromWatchlist(pageTitle)
+                } else {
+                    WatchlistAnalyticsHelper.logAddedToWatchlist(pageTitle)
+                }
+            }
+
+            val whichMessage = if (unwatch) {
+                if (isTalkPage == true) "removedwatchtext-talk" else "removedwatchtext"
+            } else {
+                if (expiry == WatchlistExpiry.NEVER) {
+                    if (isTalkPage == true) "addedwatchindefinitelytext-talk" else "addedwatchindefinitelytext"
+                } else {
+                    if (isTalkPage == true) "addedwatchexpirytext-talk" else "addedwatchexpirytext"
+                }
+            }
```

Comment:
Minor: can we have a logic below `whichMessage` for the `-talk` string and update the if statements a little bit to improve the readability?

```
if (isTalkPage == true) {
    whichMessage += "-talk"
}
```

Reply:
excellent